contracts/ics721/src/contract_test.rs

```diff
@@ -0,0 +1,634 @@
+                value: "stars1zedxv25ah8fksmg2lzrndrpkvsjqgk4zt5ff7n".into(),
+            },
+            Attribute {
+                key: "class_id".into(),
+                value: "cosmos2contract".into(),
+            },
+            Attribute {
+                key: "token_ids".into(),
+                value: "1,2,3".into(),
+            },
+        ];
+        assert_eq!(result.unwrap().attributes, expected_result);
+    }
+
+    #[test]
+    fn test_execute_receive_success() {
+        let mut deps = setup(&["channel-3", "channel-7"]);
+        test_setup(
+            deps.as_ref(),
+            "channel-3".to_string(),
+            "channel-7".to_string(),
+        );
+
+        let sender_address_str = "wasm1fucynrfkrt684pm8jrt8la5h2csvs5cnldcgqc";
+        let sender_address: Addr = Addr::unchecked(sender_address_str);
+
+        let transfer_msg = TransferMsg {
+            channel: "channel-3".to_string(),
+            class_id: "abc/123/collection-addr".to_string(),
+            class_uri: Some("abc/456/collection-addr".to_string()),
+            token_ids: vec!["1".to_string(), "2".to_string(), "3".to_string()],
+            token_uris: vec![
+                "https://metadata-url.com/my-metadata1".to_string(),
+                "https://metadata-url.com/my-metadata2".to_string(),
+                "https://metadata-url.com/my-metadata3".to_string(),
+            ],
+            remote_address: "stars1zedxv25ah8fksmg2lzrndrpkvsjqgk4zt5ff7n".to_string(),
+            timeout: Some(1000),
+        };
+
+        let cw721_receive_msg = Cw721ReceiveMsg {
+            sender: sender_address_str.to_string(),
+            token_id: "1".to_string(),
+            msg: to_binary(&transfer_msg).unwrap(),
+        };
+
+        let initial_funds = vec![];
+        let info_msg: MessageInfo = MessageInfo {
+            sender: sender_address,
+            funds: initial_funds,
+        };
+
+        let result = execute_receive(deps.as_mut(), mock_env(), info_msg, cw721_receive_msg);
+        let expected_result = [
+            Attribute {
+                key: "action".into(),
+                value: "transfer".into(),
+            },
+            Attribute {
+                key: "sender".into(),
+                value: "wasm1fucynrfkrt684pm8jrt8la5h2csvs5cnldcgqc".into(),
+            },
+            Attribute {
+                key: "receiver".into(),
+                value: "stars1zedxv25ah8fksmg2lzrndrpkvsjqgk4zt5ff7n".into(),
+            },
+            Attribute {
+                key: "class_id".into(),
+                value: "cosmos2contract".into(),
+            },
+            Attribute {
+                key: "token_ids".into(),
+                value: "1,2,3".into(),
+            },
+        ];
+        let result_clone = result.unwrap();
+        assert_eq!(result_clone.attributes, expected_result);
+        assert_eq!(result_clone.messages.len(), 1);
+    }
+
+    #[test]
+    fn test_execute_receive_payment_fail() {
```

Comment:
Should this be `receive_payment` or `receive_nft`?

Reply:
Changed the name to `test_execute_receive_nonpayable_fail`

the function its testing is below, note the nonpayable:
```
pub fn execute_receive(
  ...
) -> Result<Response, ContractError> {
    nonpayable(&info)?;
```
Then the error: 
```
pub fn nonpayable(info: &MessageInfo) -> Result<(), PaymentError> {
      ...
        Err(PaymentError::NonPayable {})
}
```